src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_gssapi.h"
+#include <assert.h>
+#include <string.h>
+
+static void NetSecurity_HandleError(uint32_t majorStatus, gss_buffer_t *outBufferHandle, uint32_t *outBufferLength)
+{
+    if (GSS_ERROR(majorStatus) || *outBufferHandle == nullptr || (*outBufferHandle)->length > UnsignedCast(std::numeric_limits<int>::max()))
```

Comment:
You're dereferencing a pointer without checking if it is nullptr itself. Since most of these are not passthroughs for user code, we should have ```assert(ptr != nullptr)``` at entry points where we control what is passed in. Dereferencing a pointer to check for nullptr also probably isn't what you want